kubevirt-vmware/pkg/controller/ovirtprovider/client.go

```diff
@@ -0,0 +1,245 @@
+package ovirtprovider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	kubevirtv1alpha1 "github.com/ovirt/v2v-conversion-host/kubevirt-vmware/pkg/apis/v2v/v1alpha1"
+)
+
+// Client struct holding implementation details required to interact with oVirt engine
+type Client struct {
+	conn    *ovirtsdk.Connection
+	ctx     context.Context
+	Cluster string
+}
+
+// NewClient creates new client struct based on connection details provided
+func NewClient(ctx context.Context, url string, username string, password string, cluster string, ca string) (*Client, error) {
```

Comment:
where does this cluster string come from?

Reply:
it is from the secret and with vm name allow us to uniquely identify a vm.

Reply:
We don't have the clusters for listing in the UI. It would be much easier if we had all the information in the beginning so we could provide additional filtering before we flag the detailRequest

Reply:
Having ID in the vm list should help us to get rid of the need for the cluster name.